src/main/groovy/io/gatling/gradle/GatlingPlugin.groovy

```diff
@@ -80,12 +81,52 @@ class GatlingPlugin implements Plugin<Project> {
                 gatling "io.gatling.highcharts:gatling-charts-highcharts:${p.extensions.getByType(GatlingPluginExtension).toolVersion}"
 
                 if (gatlingExt.includeMainOutput) {
-                    gatlingImplementation project.sourceSets.main.output
+                    gatlingImplementation p.sourceSets.main.output
                 }
                 if (gatlingExt.includeTestOutput) {
-                    gatlingImplementation project.sourceSets.test.output
+                    gatlingImplementation p.sourceSets.test.output
                 }
             }
+
+            if (shouldAddMavenCentral(p)) {
+                p.repositories {
+                    mavenCentral(name: "gatlingMavenCentral")
+                }
+            }
+        }
+    }
+
+    private static boolean shouldAddMavenCentral(Project project) {
+
+        if (!project.repositories.isEmpty()) {
+// repo adding should be attempted only of there's no explicitly declared repositories in build.gradle
```

Comment:
Please, indent comments as code, to be able to easily see the indentation without breaking the reader flow.

Reply:
done